### PR TITLE
fix: make BuildError serialisable by Celery in case of retry

### PR DESCRIPTION
**Pull request by SdgJlbl**
## Description

Workaround for [this Celery bug](https://github.com/celery/celery/issues/6990)

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: backend/substrapp/compute_tasks/image_builder.py [hunk 1]
```diff
@@ -114,6 +115,16 @@ def _get_entrypoint_from_dockerfile(dockerfile_dir: str) -> list[str]:
     raise compute_task_errors.BuildError("Invalid Dockerfile: Cannot find ENTRYPOINT")
 
 
+def _delete_kaniko_pod(create_pod: bool, k8s_client: kubernetes.client.CoreV1Api, pod_name: str) -> str:
+    logs = ""
+    if create_pod:
+        logs = get_pod_logs(k8s_client, pod_name, KANIKO_CONTAINER_NAME, ignore_pod_not_found=True)
+        delete_pod(k8s_client, pod_name)
+        for line in (logs or "").split("\n"):
+            logger.info(line, pod_name=pod_name)
```
Comment (oleobal) [inline backend/substrapp/compute_tasks/image_builder.py:124]: Do we want to log this line by line, rather than as a single JSON line?

**Review by oleobal — APPROVED** [review]
I don't have much more than a surface-level understanding of substrapp but to me this looks very good, many thanks!

(\#burncelery?)

File: backend/substrapp/compute_tasks/image_builder.py (hunk 1 above)
Comment (SdgJlbl) [reply backend/substrapp/compute_tasks/image_builder.py:124]: Well spotted, that might be the very reason why we have _so many_ lines of log coming from Kaniko.
I'd be in favour to try adding all in the same log, what do you think @guilhem-barthes ?

**Review by guilhem-barthes — APPROVED** [review]

**Pull request merged**